src/libraries/Microsoft.Win32.SystemEvents/tests/SystemEvents.InvokeOnEventsThread.cs

```diff
@@ -53,5 +51,31 @@ public void InvokeOnEventsThreadRunsOnSameThreadAsOtherEvents()
                 invoked.Dispose();
             }
         }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoNorServerCore))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void RegisterFromSTAThreadThatGoesAway_MessageStillDelivered()
+        {
+            RemoteExecutor.Invoke(() => // to ensure no one has registered for any events before
+            {
+                bool changing = false, changed = false;
+
+                // Register for the events on an STA thread that then immediately exits
+                var thread = new Thread(() =>
+                {
+                    SystemEvents.DisplaySettingsChanging += (o, e) => changing = true;
+                    SystemEvents.DisplaySettingsChanged += (o, e) => changed = true;
+                });
+                thread.SetApartmentState(ApartmentState.STA);
+                thread.IsBackground = true;
```

Comment:
```suggestion
```
Nit: This should not be needed.

Reply:
Yeah,  force of habit... never once wanted a foreground thread :-)